src/big/big_rational.cr

```diff
@@ -41,6 +41,22 @@ struct BigRational < Number
     initialize(num, 1)
   end
 
+  # Creates a exact representation of float as rational.
+  #
+  # It ensures that `BigRational.new(f) == f`
+  # It relies on fact, that mantissa is at most 53 bits
+  def initialize(num : Float)
+    frac, exp = Math.frexp num
+    ifrac = (frac.to_f64 * (1.to_i64 << 53).to_f64).to_i64
```

Comment:
yes, LLVM does constant calculations. But extraction of constants looks ok here.